packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -898,7 +896,7 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
         colorTween: colorTween,
         flex: _evaluateFlex(_animations[i]),
         selected: i == widget.currentIndex,
-        showSelectedLabels: widget.showSelectedLabels ?? bottomTheme.showSelectedLabels,
+        showSelectedLabels: widget.showSelectedLabels ?? bottomTheme.showSelectedLabels ?? true,
         showUnselectedLabels: widget.showUnselectedLabels ?? bottomTheme.showUnselectedLabels ?? _defaultShowUnselected,
```

Comment:
Should `_defaultShowUnselected` just be replaced with the value itself to match the line above?

Reply:
No, because _defaultShowUnselected is a getter that depends on the `BottomNavigationBar.type`